pkg/handler/common/provider/aws.go

```diff
@@ -244,19 +245,47 @@ func AWSSizes(region string, quota kubermaticv1.MachineDeploymentVMResourceQuota
 			continue
 		}
 
+		if !isValidArchitecture(architecture, i.PhysicalProcessor) {
+			continue
+		}
+
+		machineArchitecture := handlercommon.X64Architecture
+		if isARM64Architecture(i.PhysicalProcessor) {
+			machineArchitecture = handlercommon.ARM64Architecture
+		}
+
 		sizes = append(sizes, apiv1.AWSSize{
-			Name:       i.InstanceType,
-			PrettyName: i.PrettyName,
-			Memory:     i.Memory,
-			VCPUs:      i.VCPU,
-			GPUs:       i.GPU,
-			Price:      price,
+			Name:         i.InstanceType,
+			PrettyName:   i.PrettyName,
+			Memory:       i.Memory,
+			VCPUs:        i.VCPU,
+			GPUs:         i.GPU,
+			Price:        price,
+			Architecture: machineArchitecture,
 		})
+
 	}
 
 	return filterAWSByQuota(sizes, quota), nil
 }
 
+func isARM64Architecture(physicalProcessor string) bool {
+	// right now there is only one Arm-based processors: Graviton2
```

Comment:
I was hoping that https://instances.vantage.sh/ would have more info about architecture, but it seems that there is just basic info like:

```
[x86_64]
[x86_64 i386]
```

At the moment filtering by procesor names can work but we should think about finding a better way. The library was chosen some time ago, perhaps there are better ways right now.

Reply:
However, we might also filter based on the names. ARM instance names start with `A1`:

<img width="2092" alt="Zrzut ekranu 2021-04-26 o 11 33 57" src="https://user-images.githubusercontent.com/2823399/116061570-4eeccb00-a683-11eb-8c40-328483994ae9.png">

https://aws.amazon.com/ec2/instance-types/a1/

I think it would be a better option.

Reply:
I don't think it's not just `a1` family to use arm, I see also [c6](https://aws.amazon.com/ec2/instance-types/c6/), [r6](https://aws.amazon.com/ec2/instance-types/r6/), etc.

Reply:
Okay then it seems easier to use the processor name.